GDJS/tests/tests/CustomRuntimeObject.js

```diff
@@ -0,0 +1,722 @@
+    /** @type {gdjs.CustomRuntimeObject} */
+    let rootCustomObject;
+    /** @type {gdjs.CustomRuntimeObject} */
+    let topCustomObject;
+    /** @type {gdjs.RuntimeObject} */
+    let topLeftSprite;
+    /** @type {gdjs.RuntimeObject} */
+    let topRightSprite;
+    /** @type {gdjs.CustomRuntimeObject} */
+    let bottomCustomObject;
+    /** @type {gdjs.RuntimeObject} */
+    let bottomLeftSprite;
+    /** @type {gdjs.RuntimeObject} */
+    let bottomRightSprite;
+
+    const makeCustomObjectWith2ChildrenAt2Levels = (instanceContainer) => {
+      rootCustomObject = createCustomObject(instanceContainer);
+      topCustomObject = createCustomObject(rootCustomObject._instanceContainer);
+      // Child-object creation should be done in the onCreate method of custom object.
+      // TODO EBO Rewrite this test when an events-based object has initialInstances.
+      topLeftSprite = createSpriteObject(topCustomObject._instanceContainer);
+      topRightSprite = createSpriteObject(topCustomObject._instanceContainer);
+      topRightSprite.setX(64);
+
+      bottomCustomObject = createCustomObject(
+        rootCustomObject._instanceContainer
+      );
+      // Child-object creation should be done in the onCreate method of custom object.
+      // TODO EBO Rewrite this test when an events-based object has initialInstances.
+      bottomLeftSprite = createSpriteObject(
+        bottomCustomObject._instanceContainer
+      );
+      bottomRightSprite = createSpriteObject(
+        bottomCustomObject._instanceContainer
+      );
+      bottomRightSprite.setX(64);
+
+      bottomCustomObject.setY(64);
+    };
+
+    it('can return hit-boxes according to its children', async () => {
+      const runtimeGame = await gdjs.getPixiRuntimeGameWithAssets();
+      const runtimeScene = createSceneWithLayer(runtimeGame);
+      makeCustomObjectWith2ChildrenAt2Levels(runtimeScene);
+
+      expect(rootCustomObject.getWidth()).to.be(128);
+      expect(rootCustomObject.getHeight()).to.be(128);
+
+      expect(rootCustomObject.getAABB()).to.eql({
+        min: [0, 0],
+        max: [128, 128],
+      });
+
+      expect(rootCustomObject.getCenterXInScene()).to.be(64);
+      expect(rootCustomObject.getCenterYInScene()).to.be(64);
+
+      expect(rootCustomObject.getHitBoxes().length).to.be(4);
+      expect(rootCustomObject.getHitBoxes()[0].vertices).to.eql([
+        [64, 64],
+        [0, 64],
+        [64, 0],
+      ]);
+      expect(rootCustomObject.getHitBoxes()[1].vertices).to.eql([
+        [128, 64],
+        [64, 64],
+        [128, 0],
+      ]);
+      expect(rootCustomObject.getHitBoxes()[2].vertices).to.eql([
+        [64, 128],
+        [0, 128],
+        [64, 64],
+      ]);
+      expect(rootCustomObject.getHitBoxes()[3].vertices).to.eql([
+        [128, 128],
+        [64, 128],
+        [128, 64],
+      ]);
+    });
+
+    it('keeps hit-boxes up to date when its children move and push the bottom-right corner', async () => {
```

Comment:
When a child is moved, every parents up to the root is invalidated. It would only cover 1 or 2 lines but it's still important to do. I'm adding a todo.

Reply:
Actually, it seems covered by `describe('with custom objects as children'`.

Reply:
Perfect then!